src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticWorkerWithAnalyzers.cs

```diff
@@ -69,9 +74,14 @@ public void OnWorkspaceInitialized(bool isInitialized)
             }
         }
 
-        public async Task<ImmutableArray<DocumentDiagnostics>> GetDiagnostics(ImmutableArray<string> documentPaths)
+        public override async Task<ImmutableArray<DocumentDiagnostics>> GetDiagnostics(ImmutableArray<string> documentPaths)
         {
-            var documentIds = GetDocumentIdsFromPaths(documentPaths);
+            ImmutableArray<DocumentId> documentIds = (await Task.WhenAll(
+                               documentPaths
+                                   .Select(docPath => _workspace.GetDocumentsFromFullProjectModelAsync(docPath))))
```

Comment:
This is a good change. When projects are loaded on demand the documents may not be part of the workspace, since it will be waiting on a design time build.